unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/nio/channels/DatagramChannelTest.scala

```diff
@@ -0,0 +1,139 @@
+package org.scalanative.testsuite.javalib.nio.channels
```

Comment:
This evolution of the file certainly looks good enough to ship.
Well done.

Let me describe thoughts for a hypothetical future Evolution:
 1) It would be good if each method and its overloads were tested
    or at least exercised. Here `dc.getLocalAddress()` does not have a 
    Test but _is_ exercised.

     I started out trying to find if each entry point mentioned in the 
     JDK `DatagramChannel` description got exercised and found
     that quite a few are not.  (Lee, be glad of what you have and do
     not make yourself & others unhappy grasping beyond that!")

      Of particular interests are the overloads for `read() & `write()`.
       Datagrams (aka UDP) tends to pack more information into
       the second and third arguments in an attempt to get "one-shot"
       calls. I seem to remember writing some posixlib UDP tests
       and spending a lot of time ensuring that the features of UDP
       were addressed.  

  2) The good news is that, if I am reading these tests correctly,
       they neither know nor care if the underlying socket is IPv6 or
       IPv4.  That is robust in the wild, but leaves one wondering
       in a testing environment: which protocol is actually being used?
       Would the other one work as well?

        I believe that in the CI environment, the local address will
        always be IPv6.  In a world with unlimited resources, one
        would do a matrix test.
  
        In the current world, I suggest a comment such as 
        "// In CI, the transport is likely to be IPv6" 
        or you could use an Assume statement that IPv6 is available.
        One could then explicitly use "::1".

         The working assumption is that if IPv6 works, then IPv4 will also.
 
          One could take the converse approach and Assume IPv4
           and use 127.0.0.1.
           
            There are sometime some server connection differences
             beween IPv4 and IPv6.  If I remember correctly, FreeBSD
             and possibly NetBSD force servers to listen on both
             IPv6 & IPv4 if they want to receive packets on both/either.
             Most unix-like do allow listening on IPv4 if one is listening
             on IPv6.

              Details that concern only developers, people writing tests,
               and people with too much time on their hands.